packages/codebytes/src/drawers.tsx

```diff
@@ -0,0 +1,128 @@
+import { FlexBox, IconButton } from '@codecademy/gamut';
+import {
+  ArrowChevronLeftIcon,
+  ArrowChevronRightIcon,
+} from '@codecademy/gamut-icons';
+import styled from '@emotion/styled';
+import React, { useState } from 'react';
+
+const DrawerLabel = styled.span`
+  padding: 0.875rem 0.5rem;
+`;
+
+const LeftDrawerIcon = styled(ArrowChevronLeftIcon)<{ open?: boolean }>`
+  transition: transform 0.2s ease-in-out;
+`;
+const RightDrawerIcon = LeftDrawerIcon.withComponent(ArrowChevronRightIcon);
+
+const Drawer = styled(FlexBox)<{ open?: boolean; hideOnClose?: boolean }>`
+  position: relative;
+  ${({ open, hideOnClose }) => `
+    flex-basis: ${open ? '100%' : '0%'};
+    ${!open && hideOnClose ? 'visibility: hidden;' : ''}
+    transition: flex-basis 0.2s ${
+      open ? 'ease-out' : 'ease-in, visibility 0s 0.2s'
+    };
+
+    ${LeftDrawerIcon}, ${RightDrawerIcon} {
+      transform: rotateZ(${open ? '0' : '180'}deg)};
+    }
+  `}
+`;
+
+export type DrawersProps = {
+  leftChild: React.ReactNode;
+  rightChild: React.ReactNode;
+};
+
+export const Drawers: React.FC<DrawersProps> = ({ leftChild, rightChild }) => {
+  const [show, setShow] = useState<'left' | 'right' | 'both'>('both');
+
+  let buttonLabelL = 'hide code';
+  let buttonLabelR = 'hide output';
+
+  if (show === 'left') {
+    buttonLabelL = buttonLabelR = 'show output';
+  } else if (show === 'right') {
+    buttonLabelL = buttonLabelR = 'show code';
+  }
+
+  return (
+    <>
+      <FlexBox>
+        <Drawer
+          open={show !== 'right'}
```

Comment:
nit: since we see this logic a few times (lines 54, 71, 107) and the opposite on lines 76, 92, and 120, what do you think of making 2 booleans like the following under line 42 which could replace the equality logic?:

```
let isLeftOpen = true;
let isRightOpen = true;
```

then within your existing logic starting on 44 we could add:
```
 if (open === 'left') {
    ariaLabelCodeButton = ariaLabelOutputButton = 'show output';
    isRightOpen = false;
  } else if (open === 'right') {
    ariaLabelCodeButton = ariaLabelOutputButton = 'show code';
    isLeftOpen = false;
  }
```

_note this takes into account my rename nits above for  `open` and the aria label variable names_

Reply:
really appreciate this comment, @zippyzow ; I updated this logic a tiny bit, since isLeftOpen pretty much means that left can be open or both can be open. it works similarly with isRightOpen.

if i initialize them as false, then I can add specificity around which one is true, and then pass them into the `<Drawer />`, i.e. `<Drawer open={!isRightOpen}`

```js 
let isLeftOpen = false;
let isRightOpen = false;

  if (open === 'left') {
    ariaLabelCodeButton = ariaLabelOutputButton = 'show output';
    isLeftOpen = true;
  } else if (open === 'right') {
    ariaLabelCodeButton = ariaLabelOutputButton = 'show code';
    isRightOpen = true;
  }
```